### PR TITLE
Creation of Google Drive connector

### PR DESCRIPTION
**Pull request by yohplala**
@bmoscon 

- Creation of `gd.py` containing most notably `google_drive_write()` function -> flake8 ok.
- Documentation page for this connector
- `config.aml` updated with additional keywords (comments also reviewed throughout the file, to reduce number of characters per line, I hope you won't mind)
- `parquet.py` modified to make use of `google_drive_write()` function -> flake8 ok (here also, I made some small changes to reduce number of characters per line down to 80 max - hope it is ok for you

Thanks for your review (my 1st ever PR).

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -1,11 +1,19 @@
 # Cryptostore sample config file
 
-# Redis or Kafka are required. They are used to batch updates from cryptofeed and the storage medium of choice
+# Redis or Kafka are required. They are used to batch updates from cryptofeed
+# and the storage medium of choice.
 #
-# del_after_read: (redis only) toggles the removal of data from redis after it has been processed with cryptostore.
-# retention_time: (redis only) if data removal is enabled (via del_after_read) will allow retention of data in redis for N seconds.
-# socket: (redis only) allows redis connections via a unix domain socket
-# start_flush: toggles if redis/kafka should be flushed at the start. Primarily for debugging, it will flush ALL of redis/kafka
```
Comment (bmoscon) [inline config.yaml:8]: undo these changes to the default config.

File: config.yaml [hunk 2]
```diff
@@ -21,23 +29,40 @@ redis:
     retention_time: null
     start_flush: true
 
-# Data sources and data types configured under exchanges. Exchange names follow the naming scheme in cryptofeed (they
```
Comment (bmoscon) [inline config.yaml:24]: undo these changes

File: cryptostore/data/gd.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
+'''
+Copyright (C) 2018-2020  Pierre Juillard - pierre.juillard@gmail.com
```
Comment (bmoscon) [inline cryptostore/data/gd.py:2]: I'm not sure of the implications of merging in code someone else claims a copyright to. Would it prevent me from modifying it? Distributing it? etc. Its definitely fair to leave your name here (Original implementation by .... etc) and its also fine to not contribute this (you can close the PR). I'm just saying, I wont take any code anyone else claims a copyright to.

File: cryptostore/data/gd.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+from typing import Tuple
+import google.auth
+from google.oauth2 import service_account
+from googleapiclient.discovery import build
+from googleapiclient.http import MediaFileUpload
+from cryptostore.exceptions import InconsistentStorage
+
+
+def _get_drive(creds):
```
Comment (bmoscon) [inline cryptostore/data/gd.py:18]: no return type or param type?

File: cryptostore/data/gd.py [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+        p_folder_id = p_folders[0][0]
+
+    # Manage child folder.
+    c_folders = [(folder['id'], folder['name']) for folder in folders
+                 if ((folder['name'] == folder_name) and ('parents' in folder)
+                     and (p_folder_id in folder['parents']))]
+    if len(c_folders) > 1:
+        # If more than 2 folders with the same name, throw an error. We do not
+        # know which one is the right one to record data.
+        raise InconsistentStorage("At least 2 folders identified with name \
```
Comment (bmoscon) [inline cryptostore/data/gd.py:107]: i prefer long lines to lines shortened/continued with \. Its 2020, people have 4k monitors and other large resolution displays

File: cryptostore/data/gd.py [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+                              .execute()
+
+        return folder.get('id'), folder_name
+    else:
+        # Single folder found.
+
+        return folders[0]['id'], folder_name
+
+
+def google_drive_write(bucket, path, file_name, creds):
```
Comment (bmoscon) [inline cryptostore/data/gd.py:124]: missing types

File: cryptostore/data/gd.py [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+                        .with_scopes(['https://www.googleapis.com/auth/drive'])
+    else:
+        # use environment variable GOOGLE_APPLICATION_CREDENTIALS
+        creds, project = google.auth.default(
+                              scopes=['https://www.googleapis.com/auth/drive'])
+
+    return build('drive', 'v3', credentials=creds)
+
+
+def _get_folder_in_parent(drive, path) -> Tuple[str, str]:
```
Comment (bmoscon) [inline cryptostore/data/gd.py:44]: missing types

File: cryptostore/data/parquet.py [hunk 8]
```diff
@@ -11,8 +11,10 @@
 import pyarrow.parquet as pq
 
 from cryptostore.data.store import Store
-from cryptostore.data.gc import google_cloud_write, google_cloud_read, google_cloud_list
+from cryptostore.data.gc import google_cloud_write, google_cloud_read,\
```
Comment (bmoscon) [inline cryptostore/data/parquet.py:14]: long lines please

File: cryptostore/data/parquet.py [hunk 9]
```diff
@@ -75,10 +85,12 @@ def write(self, exchange, data_type, pair, timestamp):
                     file_name += f"{pair}-"
                 else:
                     print(var)
-                    raise ValueError("Invalid file format specified for parquet file")
```
Comment (bmoscon) [inline cryptostore/data/parquet.py:78]: undo

File: cryptostore/data/parquet.py [hunk 10]
```diff
@@ -75,10 +85,12 @@ def write(self, exchange, data_type, pair, timestamp):
...
+                    raise ValueError("Invalid file format specified for \
+                                     parquet file")
             file_name = file_name[:-1] + ".parquet"
         else:
-            file_name = f'{exchange}-{data_type}-{pair}-{int(timestamp)}.parquet'
+            file_name = f'{exchange}-{data_type}-{pair}-{int(timestamp)}\
```
Comment (bmoscon) [inline cryptostore/data/parquet.py:92]: undo

**Comment by bmoscon** [comment]
A few comments. Basically undo all the styling changes you made to existing code + a few others

**Comment by yohplala** [comment]
@bmoscon 
Bryant, I took into account allyour comments.
Please, can you review once again. It should be ok.
thanks for your help. Have a good evening, Bests,

**Comment by bmoscon** [comment]
@yohplala thanks, let me take a look and I'll merge once the review is completed.

File: cryptostore/data/gd.py [hunk 11]
```diff
@@ -0,0 +1,153 @@
+'''
+Copyright (C) 2018-2020  Bryant Moscon - bmoscon@gmail.com
```
Comment (bmoscon) [inline cryptostore/data/gd.py:2]: You can just leave the copyright off if you want, or you can add something about how you are the author. Or you can put on a public domain header with your name on it.

**Comment by bmoscon** [comment]
@yohplala i think its ready to go, just see the final comment and let me know what you want to do.

**Comment by yohplala** [comment]
@bmoscon  Hi Bryant, thanks for your feedback. I am ok as it is, no worries ;) You can merge. Thanks for your continued support Bryant, I really appreciate!
